app/grandchallenge/components/templatetags/civ.py

```diff
@@ -0,0 +1,59 @@
+from collections.abc import Iterable
+
+from django import template
+
+from grandchallenge.components.models import (
+    ComponentInterfaceValue,
+    InterfaceKindChoices,
+)
+
+register = template.Library()
+
+
+@register.filter
+def sort_civs(civs: Iterable[ComponentInterfaceValue]):
+    values = []
+    charts = []
+    thumbnails = []
+    images = []
+    files = []
+    residual = []
+
+    for v in civs:
+        if v.value is not None:
+            if v.interface.kind == InterfaceKindChoices.CHART:
+                charts.append(v)
+            else:
+                values.append(v)
+        elif v.file:
+            if v.interface.kind in (
+                InterfaceKindChoices.THUMBNAIL_PNG,
+                InterfaceKindChoices.THUMBNAIL_JPG,
+            ):
+                thumbnails.append(v)
+            else:
+                files.append(v)
+        elif v.image:
+            images.append(v)
+        else:
+            residual.append(v)
+
+    return [*values, *thumbnails, *charts, *files, *images, *residual]
+
+
+@register.filter
+def can_preview(component_interface_value):
+    return component_interface_value.interface.kind in [
+        InterfaceKindChoices.BOOL,
+        InterfaceKindChoices.FLOAT,
+        InterfaceKindChoices.INTEGER,
+        InterfaceKindChoices.STRING,
+    ]
+
+
+@register.filter
+def is_thumbnail_kind(component_interface_value):
```

Comment:
The kind definitions [are done on the class](https://github.com/comic/grand-challenge.org/blob/main/app/grandchallenge/components/models.py#L748), could these be moved there to prevent any split logic?

Reply:
I intentionally left them here since this is display/template logic which does not always align with model logic.

Reply:
I still think that it's better to put it on the class, no matter where it's used.